src/Lykke.Service.ClientAccountRecovery.Services/RecoveryFlowServiceFactory.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Threading.Tasks;
+using Autofac;
+using JetBrains.Annotations;
+using Lykke.Service.ClientAccountRecovery.Core;
+using Lykke.Service.ClientAccountRecovery.Core.Domain;
+using Lykke.Service.ClientAccountRecovery.Core.Services;
+using Lykke.Service.Kyc.Abstractions.Domain.Verification;
+using Lykke.Service.Kyc.Abstractions.Services;
+using Lykke.SettingsReader;
+
+namespace Lykke.Service.ClientAccountRecovery.Services
+{
+    [UsedImplicitly]
+    public class RecoveryFlowServiceFactory : IRecoveryFlowServiceFactory
+    {
+        private readonly ILifetimeScope _container;
+        private readonly IRecoveryLogRepository _repository;
+        private readonly IKycStatusService _kycStatusService;
+
+        public RecoveryFlowServiceFactory(ILifetimeScope container,
+            IRecoveryLogRepository repository,
+            IKycStatusService kycStatusService)
+        {
+            _container = container;
+            _repository = repository;
+            _kycStatusService = kycStatusService;
+        }
+
+        public async Task<IRecoveryFlowService> InitiateNew(string clientId)
+        {
+            var initialContext = new RecoveryContext
+            {
+                RecoveryId = Guid.NewGuid().ToString(),
+                ClientId = clientId,
+                KycPassed = await IsKycPassed(clientId)
+            };
+            var recoveryConditions = _container.Resolve<IReloadingManager<RecoveryConditions>>().CurrentValue;
+            var service = _container.Resolve<IRecoveryFlowService>(
```

Comment:
It's better to use "new" here - and you will be able to register this factory as single instance

Reply:
Resolving via container allows me add a new dependencies into IRecoveryFlowService without changing the factory. It's nothing bad having it transient.